synapseclient/core/utils.py

```diff
@@ -1262,52 +1265,35 @@ def print_tick(self):
         self._tick += 1
 
 
-def otel_trace_method(
-    method_to_trace_name: typing.Optional[typing.Callable[..., str]] = None
-):
+def run_and_attach_otel_context(
+    callable_function: Callable[..., R], current_context: context
+) -> R:
     """
-    Decorator to trace a method with OpenTelemetry in a sync environment. This function
-    is specifically written to be used on a method within a class.
-
-    This will pass the class instance as the first argument to the method. This allows
-    you to modify the name of the trace to include information about the class instance.
-
-    If you include `opentelemetry_context` as a keyword argument in the method signature,
-    and pass the current OpenTelemetry context to the method it will propagate the context
-    to the method. The is used in cases where we are running the method in a different
-    thread or event context than the one that started the trace.
-
-    Example: Decorating a method within a class that will be traced with OpenTelemetry.
-        Setting the trace name with a lambda function:
-
-            @otel_trace_method(method_to_trace_name=lambda self, **kwargs: f"Store: {self.name}")
-            def store(self, opentelemetry_context):
-
-    Arguments:
-        method_to_trace_name: A callable that takes the class instance as the first argument
-            and returns a string to be used as the trace name. If this is not provided,
-            the trace name will be set to the method name.
-
-    Returns:
-        A callable decorator that will trace the method with OpenTelemetry.
-    """
-
-    def decorator(func):
-        """Function decorator."""
-
-        def otel_trace_method_wrapper(self, *arg, **kwargs) -> None:
-            """Wrapper for the function to be traced."""
-            trace_name = (
-                method_to_trace_name(self, *arg, **kwargs)
-                if method_to_trace_name
-                else None
+    This is a generic function that will run a callable function and attach the passed in
+    OpenTelemetry context to the thread or context that the function is running on.
+
+    This is a hack to get around AsyncIO `run_in_executor` not propagating the context
+    to the code it's executing. When we are directly calling async functions after
```

Comment:
If we can remove this after SYNPY-1411 is complete, what does the difference between making this change now vs. later look like?

Reply:
Making the change to use this wrapper now prevents us from needing to modify the code in `client.py` to manually propagate the OTEL context.

Making the change later means we will have had to propagate the OTEL to the rest of the sync methods we want to call in `client.py` that haven't been called yet.